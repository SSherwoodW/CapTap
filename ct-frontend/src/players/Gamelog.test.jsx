import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { render, screen, act } from '@testing-library/react';
import GameLog from "./Gamelog";

describe("Gamelog", () => {
    it("renders the component", () => {
        render(<GameLog boxscores={[]} />)
    });

    it('matches snapshot', () => {
        const gameLogPage = render(
            <GameLog boxscores={[]}/>
        )
        expect(gameLogPage).toMatchSnapshot();
    });

    it('renders up to 5 game rows when boxscores length is less than or equal to 5', () => {
        const boxscores = [
        { gamedate: '2022-01-01', opposing_team: 'Opponent 1', minutes: '30:00', points: 20, rebounds: 10, assists: 5, turnovers: 3, steals: 2, blocks: 1, threepointsmade: 2, threepointsattempted: 5 },
        { gamedate: '2022-01-02', opposing_team: 'Opponent 2', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        ];

        render(<GameLog boxscores={boxscores} />);
        
        expect(screen.getByText('Opponent 1')).toBeDefined();

        // see if 'showmore' button is not displayed
        const button = screen.queryByRole('button')
        expect(button).toBeNull();
    });

    it('renders "Show More" button when boxscores length is greater than 5', () => {
        const boxscores = [
        { gamedate: '2022-01-01', opposing_team: 'Opponent 1', minutes: '30:00', points: 20, rebounds: 10, assists: 5, turnovers: 3, steals: 2, blocks: 1, threepointsmade: 2, threepointsattempted: 5 },
        { gamedate: '2022-01-02', opposing_team: 'Opponent 2', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-03', opposing_team: 'Opponent 3', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-04', opposing_team: 'Opponent 4', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-05', opposing_team: 'Opponent 5', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-06', opposing_team: 'Opponent 6', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        ];

        render(<GameLog boxscores={boxscores} />);
        
        expect(screen.getByText('Opponent 1')).toBeDefined();

        // see if 'showmore' button is displayed
        const button = screen.queryByRole('button')
        expect(button).toBeDefined();
    });

    it('toggles "Show More" button and displays more gamelogs when clicked', () => {
        const boxscores = [
        { gamedate: '2022-01-01', opposing_team: 'Opponent 1', minutes: '30:00', points: 20, rebounds: 10, assists: 5, turnovers: 3, steals: 2, blocks: 1, threepointsmade: 2, threepointsattempted: 5 },
        { gamedate: '2022-01-02', opposing_team: 'Opponent 2', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-03', opposing_team: 'Opponent 3', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-04', opposing_team: 'Opponent 4', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-05', opposing_team: 'Opponent 5', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        { gamedate: '2022-01-06', opposing_team: 'Opponent 6', minutes: '25:00', points: 15, rebounds: 5, assists: 8, turnovers: 2, steals: 3, blocks: 0, threepointsmade: 1, threepointsattempted: 3 },
        ];

        render(<GameLog boxscores={boxscores} />);
        screen.debug();
        expect(screen.queryByText('Opponent 6')).toBeNull();

        const showMoreButton = screen.getByText('Show More');
        act(() => {
            showMoreButton.click()
        })

        expect(screen.queryByText('Opponent 6')).toBeDefined();
        expect(showMoreButton.textContent).toBe('Show Less');
    })
})