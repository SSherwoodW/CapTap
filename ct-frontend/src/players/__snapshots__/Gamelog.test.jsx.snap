// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gamelog > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid bg-gray-200 rounded-sm p-4 mb-4 text-center overflow-x-auto max-w-full"
      >
        <h3
          class="text-xl bg-indigo-500 text-white mb-2 p-2 rounded-t-sm"
        >
          Game Log
        </h3>
        <div
          class="border-b-2 border-indigo-600 mb-2"
        />
        <div
          class="grid grid-cols-10 bg-gray-300 p-3 "
        >
          <div
            class="text-sm font-bold text-gray-700"
          >
            Date
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Opponent
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Minutes
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Points
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Rebounds
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Assists
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Turnovers
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Steals
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            Blocks
          </div>
          <div
            class="text-sm font-bold text-gray-700"
          >
            3pm/att
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid bg-gray-200 rounded-sm p-4 mb-4 text-center overflow-x-auto max-w-full"
    >
      <h3
        class="text-xl bg-indigo-500 text-white mb-2 p-2 rounded-t-sm"
      >
        Game Log
      </h3>
      <div
        class="border-b-2 border-indigo-600 mb-2"
      />
      <div
        class="grid grid-cols-10 bg-gray-300 p-3 "
      >
        <div
          class="text-sm font-bold text-gray-700"
        >
          Date
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Opponent
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Minutes
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Points
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Rebounds
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Assists
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Turnovers
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Steals
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          Blocks
        </div>
        <div
          class="text-sm font-bold text-gray-700"
        >
          3pm/att
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
